### PR TITLE
feat: search product by name

### PR DESCRIPTION
**Pull request by dsitien**

### CODE REVIEW

File: GearShop/DataAccess/DAO/ProductDAO.cs [hunk 1]
```diff
@@ -85,5 +86,39 @@ public static async Task<List<ProductModel>> GetProductListAdmin()
...
+            try
+            {
+                var dbContext = new ProductContext();
+                products = await dbContext.Products.Where(p => p.ProName.Contains(productName)).ToListAsync();
+                if (products != null)
+                {
+                    foreach (var product in products)
+                    {
+
+
```
Comment (cleny30) [inline GearShop/DataAccess/DAO/ProductDAO.cs:104]: xoá Enter

File: GearShop/DataAccess/DAO/ProductDAO.cs [hunk 2]
```diff
@@ -85,5 +86,39 @@ public static async Task<List<ProductModel>> GetProductListAdmin()
...
+                        model.Add(new ProductData
```
Comment (cleny30) [inline GearShop/DataAccess/DAO/ProductDAO.cs:105]: Dùng CopyProperties

File: GearShop/WebClient/Controllers/ShopController.cs [hunk 3]
```diff
@@ -51,5 +51,32 @@ public async Task<IActionResult> Index()
             ShopModel data = _shopService.GetData(productList, brandList, categoryList, sortFilter, orderFilter, category, brand, currentPage);
             return View(data);
         }
+
+
+        [HttpGet]
+        public async Task<IActionResult> Search(string pattern)
```
Comment (cleny30) [inline GearShop/WebClient/Controllers/ShopController.cs]: Viết ở SearchController

File: GearShop/DataAccess/DAO/ProductDAO.cs (hunk 2 above)
Comment (cleny30) [inline GearShop/DataAccess/DAO/ProductDAO.cs]: Dùng CopyProperties

File: GearShop/DataAccess/DAO/ProductDAO.cs [hunk 4]
```diff
@@ -85,5 +86,39 @@ public static async Task<List<ProductModel>> GetProductListAdmin()
...
+            List<ProductData> model = new List<ProductData>();
+            try
+            {
+                var dbContext = new ProductContext();
+                products = await dbContext.Products.Where(p => p.ProName.Contains(productName)).ToListAsync();
+                if (products != null)
+                {
+                    foreach (var product in products)
+                    {
+
```
Comment (cleny30) [inline GearShop/DataAccess/DAO/ProductDAO.cs]: xoá Enter (optional)

File: GearShop/WebClient/Controllers/ShopController.cs (hunk 3 above)
Comment (dsitien) [reply GearShop/WebClient/Controllers/ShopController.cs]: done

File: GearShop/DataAccess/DAO/ProductDAO.cs (hunk 2 above)
Comment (dsitien) [reply GearShop/DataAccess/DAO/ProductDAO.cs]: done

File: GearShop/DataAccess/DAO/ProductDAO.cs (hunk 4 above)
Comment (dsitien) [reply GearShop/DataAccess/DAO/ProductDAO.cs]: done

**Pull request merged**